Mathlib/CategoryTheory/Category/ReflQuiv.lean

```diff
@@ -75,6 +75,50 @@ theorem forgetToQuiv.Faithful : Functor.Faithful (forgetToQuiv) where
 
 theorem forget_forgetToQuiv : forget ⋙ forgetToQuiv = Quiv.forget := rfl
 
+/-- An isomorphism of quivers lifts to an isomorphism of reflexive quivers given a suitable
+compatibility with the identities. -/
+def isoOfQuivIso {V W : Type u} [ReflQuiver.{v + 1} V] [ReflQuiver.{v + 1} W]
+    (e : Quiv.of V ≅ Quiv.of W)
+    (h_id : ∀ (X : V), e.hom.map (𝟙rq X) = ReflQuiver.id (obj := W) (e.hom.obj X)) :
+    ReflQuiv.of V ≅ ReflQuiv.of W where
+  hom := ReflPrefunctor.mk e.hom h_id
+  inv := by
+    refine ReflPrefunctor.mk e.inv ?_
+    intro Y
+    have : Quiver.homOfEq (e.inv.map (e.hom.map _)) _ _ = (𝟭q _).map _ :=
+      Prefunctor.congr_hom e.hom_inv_id (ReflQuiver.id (obj := V) (e.inv.obj Y))
+    simp only [Prefunctor.id_obj, Prefunctor.id_map] at this
+    rw [congrArg e.inv.map (h_id (e.inv.obj Y))] at this
+    rw [← this]
+    have hY : e.hom.obj (e.inv.obj Y) = Y := by
+      rw [← Prefunctor.comp_obj]
+      show (e.inv ≫ e.hom).obj _ = _
+      rw [e.inv_hom_id, Quiv.id_eq_id, Prefunctor.id_obj]
+    rw [← Prefunctor.homOfEq_map _ _ hY hY]
+    simp only [of_val, ReflQuiver.homOfEq_id]
```

Comment:
After adding some API for isomorphisms in `Quiv`:
```lean
namespace Quiv

variable {V W : Quiv} (e : V ≅ W)

@[simps]
def equivOfIso : V ≃ W where
  toFun := e.hom.obj
  invFun := e.inv.obj
  left_inv := Prefunctor.congr_obj e.hom_inv_id
  right_inv := Prefunctor.congr_obj e.inv_hom_id

@[simp]
lemma inv_obj_hom_obj_of_iso (X : V) :
    e.inv.obj (e.hom.obj X) = X :=
  (equivOfIso e).left_inv X

@[simp]
lemma hom_obj_inv_obj_of_iso (Y : W) :
    e.hom.obj (e.inv.obj Y) = Y :=
  (equivOfIso e).right_inv Y

lemma hom_map_inv_map_of_iso {V W : Quiv} (e : V ≅ W) {X Y : W} (f : X ⟶ Y) :
    e.hom.map (e.inv.map f) =
      Quiver.homOfEq f (by simp) (by simp) := by
  rw [← Prefunctor.comp_map]
  exact (Prefunctor.congr_hom e.inv_hom_id.symm f).symm

lemma inv_map_hom_map_of_iso {V W : Quiv} (e : V ≅ W) {X Y : V} (f : X ⟶ Y) :
    e.inv.map (e.hom.map f) =
      Quiver.homOfEq f (by simp) (by simp) :=
  hom_map_inv_map_of_iso e.symm f

@[simps]
def homEquivOfIso {V W : Quiv} (e : V ≅ W) {X Y : V} :
    (X ⟶ Y) ≃ (e.hom.obj X ⟶ e.hom.obj Y) where
  toFun f := e.hom.map f
  invFun g := Quiver.homOfEq (e.inv.map g) (by simp) (by simp)
  left_inv f := by simp [inv_map_hom_map_of_iso]
  right_inv g := by simp [hom_map_inv_map_of_iso]

end Quiv
```
The proof here becomes:
```suggestion
def isoOfQuivIso {V W : Type u} [ReflQuiver.{v + 1} V] [ReflQuiver.{v + 1} W]
    (e : Quiv.of V ≅ Quiv.of W)
    (h_id : ∀ (X : V), e.hom.map (𝟙rq X) = ReflQuiver.id (obj := W) (e.hom.obj X)) :
    ReflQuiv.of V ≅ ReflQuiv.of W where
  hom := ReflPrefunctor.mk e.hom h_id
  inv := ReflPrefunctor.mk e.inv
    (fun Y ↦ (Quiv.homEquivOfIso e).injective
      (by simp [Quiv.hom_map_inv_map_of_iso, h_id]))
```

Reply:
Thanks, implemented.